src/features/staking/StakingModal.tsx

```diff
@@ -1,188 +1,157 @@
-} from "./utils/helpers";
-
-interface StakingModalInterface {
-  stakingModalRef: RefObject<HTMLDialogElement>;
-  chainId: number;
-}
-
-export const StakingModal: FC<StakingModalInterface> = ({
-  stakingModalRef,
-  chainId,
-}) => {
-  const [stakeOrUnstake, setStakeOrUnstake] = useState<StakeOrUnstake>(
-    StakeOrUnstake.STAKE,
-  );
+import { TxType, useStakingModalStore } from "./store/useStakingModalStore";
+import { AmountStakedText } from "./subcomponents/AmountStakedText";
+import { actionButtonsObject } from "./utils/actionButtonObject";
+import { modalButtonActionsAndText } from "./utils/modalButtonActionsAndText";
+import { modalTxLoadingStateHeadlines } from "./utils/modalTxLoadingStateHeadlines";
+import { transactionTrackerMessages } from "./utils/transactionTrackerMesssages";
+
+export const StakingModal = () => {
+  const { setShowStakingModal, txType } = useStakingModalStore();
 
   const formReturn = useForm();
-  const { watch, setValue } = formReturn;
-  const stakingAmount = Number(watch("stakingAmount") || "0");
+  const { watch } = formReturn;
+  const stakingAmount = watch("stakingAmount") || "0";
+
+  const { astAllowanceFormatted: astAllowance } = useAstAllowance();
 
-  const { astAllowanceRaw } = useAstAllowance();
   const {
-    unstakableSAstBalanceRaw: unstakableSAstBalance,
-    astBalanceRaw: astBalance,
+    unstakableSAstBalanceFormatted: unstakableSastBalance,
+    astBalanceFormatted: astBalance,
   } = useTokenBalances();
 
   const needsApproval =
-    stakeOrUnstake === StakeOrUnstake.STAKE &&
-    stakingAmount > 0 &&
-    astAllowanceRaw < stakingAmount;
+    txType === TxType.STAKE &&
+    Number(stakingAmount) > 0 &&
+    Number(astAllowance) < Number(stakingAmount);
 
-  const canStake =
-    !needsApproval && stakingAmount <= astBalance && stakingAmount > 0;
+  const canStake = txType === TxType.STAKE && !needsApproval;
 
   const canUnstake =
-    stakeOrUnstake === StakeOrUnstake.UNSTAKE && stakingAmount > 0
-      ? stakingAmount <= unstakableSAstBalance
-      : false;
+    stakingAmount <= Number(unstakableSastBalance) && txType === TxType.UNSTAKE;
 
-  const { approve, statusApprove } = useApproveAst({
+  const { approveAst, dataApproveAst, resetApproveAst } = useApproveAst({
     stakingAmount,
     enabled: needsApproval,
   });
 
-  const { stake, resetStake, transactionReceiptStake, statusStake } =
-    useStakeAst({
-      stakingAmount,
-      enabled: canStake,
-    });
+  const { stakeAst, resetStakeAst, dataStakeAst } = useStakeAst({
+    stakingAmount,
+    enabled: canStake,
+  });
 
-  const { unstake, resetUnstake, statusUnstake, transactionReceiptUnstake } =
-    useUnstakeSast({
-      unstakingAmount: stakingAmount,
-      canUnstake,
-    });
+  const { unstakeSast, resetUnstakeSast, dataUnstakeSast } = useUnstakeSast({
+    unstakingAmount: stakingAmount,
+    canUnstake,
+  });
 
-  const buttonText = buttonStatusText({
-    stakeOrUnstake,
+  const transactionHashes =
```

Comment:
Given this is only one hash, I would suggest calling it something like `currentTransactionHash` to be a little clearer.